spikeextractors/extractors/bindatrecordingextractor/bindatrecordingextractor.py

```diff
@@ -77,66 +135,96 @@ def get_traces(self, channel_ids=None, start_frame=None, end_frame=None, return_
             exp_idx = self._dtype.find('int') + 3
             exp = int(self._dtype[exp_idx:])
             recordings = recordings.astype('float32') - 2**(exp - 1)
-        if self._gain is not None and return_scaled:
-            recordings = recordings * self._gain
+        if 'gain' in self.get_shared_channel_property_names() and return_scaled:
+            recordings = recordings * np.reshape(self.get_channel_gains(channel_ids=channel_ids), (len(channel_ids), 1))
```

Comment:
@CodyCBakerPhD a nicer way is the following:
```suggestion
            recordings = recordings * self.get_channel_gains(channel_ids=channel_ids)[:, None]
```

Reply:
Ahh, very nice!